### DISCUSSION TITLE
Allow environment variables to be used as a color in a palette

### DISCUSSION THREAD

**Discussion by sigint-run** [JanDeDobbeleer/oh-my-posh#5360] [Ideas]
So using environment variables in palettes. Maybe it is already possible, but it seems that .Env.VarName is only working for templates. 

But if we could do something like:
```
"palette" :{
  "someColor": ${VAR_NAME},
  ...
}
```
then we could use it normally in a segment like this:
```
{
  ...
  "foreground": "p:someColor"
}
```

and having somewhere in a script or .bashrc / .zshrc:
```
VAR_NAME="#abc123"
```

that would allow to use environment variables for different sessions on different terminals, or to have matching color themes with other programs like tmux.

How about that? :)

**Comment by JanDeDobbeleer** [comment]
@sigint-run this is why `palettes` was designed though. Would that fit the use case?

**Reply by sigint-run** [reply]
I apologize, I created a new comment instead of replying to your question.

**Comment by sigint-run** [comment]
Palettes allows me to make themes really easily: I can change a color in the palette, and there you go: a new theme! So palettes are great as is.

My suggestion was to allow to use an environment variable as a color in the palette, so it can be even more flexible. I could have the same oh-my-posh .json file on multiple systems, and yet, the theme colors would not be the same because it will use the local color environment variable on that system . Also, I could use the same variable on other programs, like tmux, neovim etc. and by changing that variable, they would all match the oh-my-posh's theme instantly.

The key here is that I would use one single oh-my-posh file for every system I use so the segments match, but I could still use that single file and have different color themes. Plus use those colors in other tools I use for esthetic reasons.
